src/hyperlight_guest_bin/src/lib.rs

```diff
@@ -139,11 +141,50 @@ pub static mut OS_PAGE_SIZE: u32 = 0;
 // to satisfy the clippy when cfg == test
 #[allow(dead_code)]
 fn panic(info: &core::panic::PanicInfo) -> ! {
-    let msg = info.to_string();
-    let c_string = alloc::ffi::CString::new(msg)
-        .unwrap_or_else(|_| alloc::ffi::CString::new("panic (invalid utf8)").unwrap());
+    _panic_handler(info)
+}
+
+static PANIC_BUF: Mutex<String<512>> = Mutex::new(String::new());
+
+#[inline(always)]
+fn _panic_handler(info: &core::panic::PanicInfo) -> ! {
+    let mut panic_buf_guard = PANIC_BUF.lock();
```

Comment:
Is there any chance we can release the guard before exiting the vm? I am worried about what would happen if we panic on next entry, and the lock is already held

Reply:
Per offline conversation, we'll stack allocate the buffer to avoid needing a mutex at all